### PR TITLE
[PM-13706] Add repository + stored procedures for private key regeneration

### PR DESCRIPTION
**Pull request by Thomas-Avery**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-13706

## 📔 Objective

The purpose of this PR is to add the database stored procedure and repository layer for the private key regeneration project.

This initial phase will only target users not in organizations and without emergency access setup.
In future PRs, the database operations will expand to handle these cases.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: src/Infrastructure.Dapper/KeyManagement/Repositories/UserAsymmetricKeysRepository.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        connectionString, readOnlyConnectionString)
+    {
+    }
+
+    public async Task RegenerateUserAsymmetricKeysAsync(UserAsymmetricKeys userAsymmetricKeys)
+    {
+        await using var connection = new SqlConnection(ConnectionString);
+
+        await connection.ExecuteAsync("[dbo].[UserAsymmetricKeys_Regenerate]",
+            userAsymmetricKeys, commandType: CommandType.StoredProcedure);
```
Comment (mzieniukbw) [inline src/Infrastructure.Dapper/KeyManagement/Repositories/UserAsymmetricKeysRepository.cs:29]: The last parameter of `UserAsymmetricKeys_Regenerate` is named `PrivateKey`, but there is no equivalently named field in `UserAsymmetricKeys`. There is `UserKeyEncryptedPrivateKey`.

**Review by mzieniukbw — CHANGES_REQUESTED** [review]

File: src/Infrastructure.Dapper/KeyManagement/Repositories/UserAsymmetricKeysRepository.cs (hunk 1 above)
Comment (Thomas-Avery) [reply src/Infrastructure.Dapper/KeyManagement/Repositories/UserAsymmetricKeysRepository.cs]: Great catch fixed in https://github.com/bitwarden/server/pull/4898/commits/9278677d3065435027ea97c22ba81266fd33e99a

**Review by mzieniukbw — APPROVED** [review]
LGTM

**Review by mzieniukbw — APPROVED** [review]

**Review by rkac-bw — APPROVED** [review]
lgtm

**Review by mzieniukbw — APPROVED** [review]

**Review by rkac-bw — APPROVED** [review]
lgtm

**Pull request merged**